From patchwork Wed Apr 28 10:18:41 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>
X-Patchwork-Id: 12228503
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-18.8 required=3.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,INCLUDES_CR_TRAILER,
	INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,USER_AGENT_GIT
	autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id D9086C433B4
	for <kvm@archiver.kernel.org>; Wed, 28 Apr 2021 10:18:52 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id A894F61431
	for <kvm@archiver.kernel.org>; Wed, 28 Apr 2021 10:18:52 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S239185AbhD1KTg (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Wed, 28 Apr 2021 06:19:36 -0400
Received: from lindbergh.monkeyblade.net ([23.128.96.19]:34522 "EHLO
        lindbergh.monkeyblade.net" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S236343AbhD1KTf (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 28 Apr 2021 06:19:35 -0400
Received: from mail-wm1-x32f.google.com (mail-wm1-x32f.google.com
 [IPv6:2a00:1450:4864:20::32f])
        by lindbergh.monkeyblade.net (Postfix) with ESMTPS id 84333C06138B
        for <kvm@vger.kernel.org>; Wed, 28 Apr 2021 03:18:50 -0700 (PDT)
Received: by mail-wm1-x32f.google.com with SMTP id g65so4306726wmg.2
        for <kvm@vger.kernel.org>; Wed, 28 Apr 2021 03:18:50 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=linaro.org; s=google;
        h=from:to:cc:subject:date:message-id:in-reply-to:references
         :mime-version:content-transfer-encoding;
        bh=W/65gl/q8IPHxxfmkIEqzGwlIffS1aBvU9xDbEnZyHM=;
        b=SSrrTtwZJ02Zr0JA6NRj0GmP0uyFPOUSFabhokoV6Sy5GNCyllTC4VWQMLE81igmUe
         Wh0Vi6utYveAA/fpekiI7G8ty768Nqo3K2LOp81FzggyzgrXoRp1v5KCSAMdAMjCuyuX
         r9IM1GiXmva9cV/s22qi9+5GmGGbvx5LX8EDykKoW42J+NlR2k9rKWnZXSzadxTkzeUd
         8J/bswhnJIBlc10SW99J5bPaBrR8JUigyAKh1GY1S03X1swxtLM4p+pB0ZXIwfxYh19v
         CSzoeIhOjW5U/V0rLWA1tjAH0VlGLBK6V5mrPM0R1DcsKn2suEmABcoaH6S5DiU8qNgM
         REDA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=W/65gl/q8IPHxxfmkIEqzGwlIffS1aBvU9xDbEnZyHM=;
        b=EyXLYwaApc/BnKMcdUWfNG7cEoLTuikEDK4l8CkIS5S91eH8/0nmGX1q/WMGEXmagR
         V6nhu6neKm66JKU8OTuFMFOH/E0MC6pf3qvsN8q521hz7Ya0A2YaC10J72yhVvjU8aIX
         m4t5VWCAKn7B8iLAGLXVfp0HNjanFoY82fVHAndBVe1i/QdSxuKAOkjA7Zmmvf97SxFB
         zZgQ59f5PY4CvBgJXKI980wfRhmMCWs0nwQqzMbk3wYI6vvfku6CJ1vHX8shwzoX6IhI
         9RCArz06WaakPnYdVm8lzD8kZoGw1/HghkKVBkBgD6IBOSqGCZOdX+e5yf8ZhJsqy/H9
         39Zg==
X-Gm-Message-State: AOAM530nmbKpqhr30SMa18rGysyfcDLlPf22vNTcN84SqjrCtPkAYrxV
        gPQxEFKTUr+a6Bff3UXfxQ2f6g==
X-Google-Smtp-Source: 
 ABdhPJxRO3MjPBAuuf7/5yeRHIOMTjCLbzPgeFWy3MG9Ge5F6gbUkBS3K7ic1Vpc5z70bIjpSShKyA==
X-Received: by 2002:a7b:cd85:: with SMTP id y5mr3586309wmj.93.1619605129193;
        Wed, 28 Apr 2021 03:18:49 -0700 (PDT)
Received: from zen.linaroharston ([51.148.130.216])
        by smtp.gmail.com with ESMTPSA id
 b14sm7950544wrf.75.2021.04.28.03.18.44
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Wed, 28 Apr 2021 03:18:44 -0700 (PDT)
Received: from zen.lan (localhost [127.0.0.1])
        by zen.linaroharston (Postfix) with ESMTP id 27D401FF87;
        Wed, 28 Apr 2021 11:18:44 +0100 (BST)
From: =?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>
To: kvm@vger.kernel.org
Cc: shashi.mallela@linaro.org, alexandru.elisei@arm.com,
 eric.auger@redhat.com, qemu-arm@nongnu.org,
 linux-arm-kernel@lists.infradead.org, kvmarm@lists.cs.columbia.edu,
 christoffer.dall@arm.com, maz@kernel.org,
 =?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>
Subject: [kvm-unit-tests PATCH v1 1/4] arm64: split its-trigger test into KVM
 and TCG variants
Date: Wed, 28 Apr 2021 11:18:41 +0100
Message-Id: <20210428101844.22656-2-alex.bennee@linaro.org>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20210428101844.22656-1-alex.bennee@linaro.org>
References: <20210428101844.22656-1-alex.bennee@linaro.org>
MIME-Version: 1.0
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

A few of the its-trigger tests rely on IMPDEF behaviour where caches
aren't flushed before invall events. However TCG emulation doesn't
model any invall behaviour and as we can't probe for it we need to be
told. Split the test into a KVM and TCG variant and skip the invall
tests when under TCG.

Signed-off-by: Alex Bennée <alex.bennee@linaro.org>
Cc: Shashi Mallela <shashi.mallela@linaro.org>
---
 arm/gic.c         | 60 +++++++++++++++++++++++++++--------------------
 arm/unittests.cfg | 11 ++++++++-
 2 files changed, 45 insertions(+), 26 deletions(-)

diff --git a/arm/gic.c b/arm/gic.c
index 98135ef..96a329d 100644
--- a/arm/gic.c
+++ b/arm/gic.c
@@ -36,6 +36,7 @@ static struct gic *gic;
 static int acked[NR_CPUS], spurious[NR_CPUS];
 static int irq_sender[NR_CPUS], irq_number[NR_CPUS];
 static cpumask_t ready;
+static bool under_tcg;
 
 static void nr_cpu_check(int nr)
 {
@@ -734,32 +735,38 @@ static void test_its_trigger(void)
 	/*
 	 * re-enable the LPI but willingly do not call invall
 	 * so the change in config is not taken into account.
-	 * The LPI should not hit
+	 * The LPI should not hit. This does however depend on
+	 * implementation defined behaviour - under QEMU TCG emulation
+	 * it can quite correctly process the event directly.
 	 */
-	gicv3_lpi_set_config(8195, LPI_PROP_DEFAULT);
-	stats_reset();
-	cpumask_clear(&mask);
-	its_send_int(dev2, 20);
-	wait_for_interrupts(&mask);
-	report(check_acked(&mask, -1, -1),
-			"dev2/eventid=20 still does not trigger any LPI");
-
-	/* Now call the invall and check the LPI hits */
-	stats_reset();
-	cpumask_clear(&mask);
-	cpumask_set_cpu(3, &mask);
-	its_send_invall(col3);
-	wait_for_interrupts(&mask);
-	report(check_acked(&mask, 0, 8195),
-			"dev2/eventid=20 pending LPI is received");
-
-	stats_reset();
-	cpumask_clear(&mask);
-	cpumask_set_cpu(3, &mask);
-	its_send_int(dev2, 20);
-	wait_for_interrupts(&mask);
-	report(check_acked(&mask, 0, 8195),
-			"dev2/eventid=20 now triggers an LPI");
+	if (under_tcg) {
+		report_skip("checking LPI triggers without invall");
+	} else {
+		gicv3_lpi_set_config(8195, LPI_PROP_DEFAULT);
+		stats_reset();
+		cpumask_clear(&mask);
+		its_send_int(dev2, 20);
+		wait_for_interrupts(&mask);
+		report(check_acked(&mask, -1, -1),
+		       "dev2/eventid=20 still does not trigger any LPI");
+
+		/* Now call the invall and check the LPI hits */
+		stats_reset();
+		cpumask_clear(&mask);
+		cpumask_set_cpu(3, &mask);
+		its_send_invall(col3);
+		wait_for_interrupts(&mask);
+		report(check_acked(&mask, 0, 8195),
+		       "dev2/eventid=20 pending LPI is received");
+
+		stats_reset();
+		cpumask_clear(&mask);
+		cpumask_set_cpu(3, &mask);
+		its_send_int(dev2, 20);
+		wait_for_interrupts(&mask);
+		report(check_acked(&mask, 0, 8195),
+		       "dev2/eventid=20 now triggers an LPI");
+	}
 
 	report_prefix_pop();
 
@@ -981,6 +988,9 @@ int main(int argc, char **argv)
 	if (argc < 2)
 		report_abort("no test specified");
 
+	if (argc == 3 && strcmp(argv[2], "tcg") == 0)
+		under_tcg = true;
+
 	if (strcmp(argv[1], "ipi") == 0) {
 		report_prefix_push(argv[1]);
 		nr_cpu_check(2);
diff --git a/arm/unittests.cfg b/arm/unittests.cfg
index f776b66..c72dc34 100644
--- a/arm/unittests.cfg
+++ b/arm/unittests.cfg
@@ -184,13 +184,22 @@ extra_params = -machine gic-version=3 -append 'its-introspection'
 groups = its
 arch = arm64
 
-[its-trigger]
+[its-trigger-kvm]
 file = gic.flat
 smp = $MAX_SMP
+accel = kvm
 extra_params = -machine gic-version=3 -append 'its-trigger'
 groups = its
 arch = arm64
 
+[its-trigger-tcg]
+file = gic.flat
+smp = $MAX_SMP
+accel = tcg
+extra_params = -machine gic-version=3 -append 'its-trigger tcg'
+groups = its
+arch = arm64
+
 [its-migration]
 file = gic.flat
 smp = $MAX_SMP

From patchwork Wed Apr 28 10:18:42 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>
X-Patchwork-Id: 12228501
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-18.8 required=3.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,INCLUDES_CR_TRAILER,
	INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,USER_AGENT_GIT
	autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 7116BC43600
	for <kvm@archiver.kernel.org>; Wed, 28 Apr 2021 10:18:51 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 3FACE61435
	for <kvm@archiver.kernel.org>; Wed, 28 Apr 2021 10:18:51 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S239230AbhD1KTe (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Wed, 28 Apr 2021 06:19:34 -0400
Received: from lindbergh.monkeyblade.net ([23.128.96.19]:34510 "EHLO
        lindbergh.monkeyblade.net" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S239186AbhD1KTd (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 28 Apr 2021 06:19:33 -0400
Received: from mail-wr1-x42d.google.com (mail-wr1-x42d.google.com
 [IPv6:2a00:1450:4864:20::42d])
        by lindbergh.monkeyblade.net (Postfix) with ESMTPS id 1865AC06138A
        for <kvm@vger.kernel.org>; Wed, 28 Apr 2021 03:18:48 -0700 (PDT)
Received: by mail-wr1-x42d.google.com with SMTP id h4so53384776wrt.12
        for <kvm@vger.kernel.org>; Wed, 28 Apr 2021 03:18:48 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=linaro.org; s=google;
        h=from:to:cc:subject:date:message-id:in-reply-to:references
         :mime-version:content-transfer-encoding;
        bh=JIGl3K3qE3k2BXQSfM80xkGe4hRnH3OA6tpOjjNrd3E=;
        b=Tocu2ZsTcx7VJN6+MVikFztL2wEkgN20wGnOqWs/UwSTwQ8IT1wNoZoLCbL1uytn3f
         qCa6ykXIJ39hLu4V+KCW1lOlOmcToymqvCX7t0lR3o87WapMMqa8pTdRKzo+7J5B3GdX
         cdiIdktldMin4hXEw4px5yAX0xr90WWmxO8smtcZti/I1/LApf837ST2k6Au1cXg6Z5y
         ZmqDDGg96rXNe6zD9MZlfwQRtM+smRh0A5U6Y4j93mTVpC8y5NGpu6NsLMBfYp2NIj7k
         1sGEevKSlWBdzR0UJhugw5fzOXmVcF5hl8CdGbdCsoEuHBke8NPjMo+982KpfVcXN8mH
         6ylw==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=JIGl3K3qE3k2BXQSfM80xkGe4hRnH3OA6tpOjjNrd3E=;
        b=figzsppJRl0xHkm9kxfkXyTEy6ZDOifpooC7AnKPKBt+nf1WVkxstLljarlOI28qGW
         ztPJn+a8aJ1V6Z35RgB3LemwnUwSMS8iHKSUoWL/j/QgBiCvd2dj7c4oCJYD+NriS9Sj
         RxO+JGe+T8Gnq9mWMoVNPZ7o3XTQGRGRTuOLMdhS7GKissUs3CI4oxlG3wSoCqjfGcCu
         4zuQB92EaUx6ySKpkq3qHJ5YjnxMp6RFtmWK+xBiMe0635u3dZ9Ug5GT7g0Ovq9XO1ft
         1xqCSHtGoXPHjfcpcDBJBSm/HcnZ+i4TS4NUKnaR5CZz1N+POb/3SYhPSW0Odzjgfe8w
         d+dg==
X-Gm-Message-State: AOAM532jwolrgDmJw4SJehGkLamNO73Zps3yZu8LXwjCjgEBJINDyn/M
        LQxd57obKFclUWJUBPS0MLEAyg==
X-Google-Smtp-Source: 
 ABdhPJzY9oNGXhOxLJCihSwIGFeQ6N62MW7ZFy5SInbOGtC3xunUNEQ7q6CYNQ8xm6AW5+ASciIkSA==
X-Received: by 2002:adf:fdcd:: with SMTP id
 i13mr34506297wrs.185.1619605126850;
        Wed, 28 Apr 2021 03:18:46 -0700 (PDT)
Received: from zen.linaroharston ([51.148.130.216])
        by smtp.gmail.com with ESMTPSA id
 s83sm3294855wms.16.2021.04.28.03.18.44
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Wed, 28 Apr 2021 03:18:44 -0700 (PDT)
Received: from zen.lan (localhost [127.0.0.1])
        by zen.linaroharston (Postfix) with ESMTP id 343B41FF8C;
        Wed, 28 Apr 2021 11:18:44 +0100 (BST)
From: =?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>
To: kvm@vger.kernel.org
Cc: shashi.mallela@linaro.org, alexandru.elisei@arm.com,
 eric.auger@redhat.com, qemu-arm@nongnu.org,
 linux-arm-kernel@lists.infradead.org, kvmarm@lists.cs.columbia.edu,
 christoffer.dall@arm.com, maz@kernel.org,
 =?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>
Subject: [kvm-unit-tests PATCH v1 2/4] scripts/arch-run: don't use deprecated
 server/nowait options
Date: Wed, 28 Apr 2021 11:18:42 +0100
Message-Id: <20210428101844.22656-3-alex.bennee@linaro.org>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20210428101844.22656-1-alex.bennee@linaro.org>
References: <20210428101844.22656-1-alex.bennee@linaro.org>
MIME-Version: 1.0
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

The very fact that QEMU drops the deprecation warning while running is
enough to confuse the its-migration test into failing. The boolean
options server and wait have accepted the long form options for a long
time.

Signed-off-by: Alex Bennée <alex.bennee@linaro.org>
---
 scripts/arch-run.bash | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/scripts/arch-run.bash b/scripts/arch-run.bash
index 5997e38..70693f2 100644
--- a/scripts/arch-run.bash
+++ b/scripts/arch-run.bash
@@ -122,14 +122,14 @@ run_migration ()
 	trap 'kill 0; exit 2' INT TERM
 	trap 'rm -f ${migout1} ${migsock} ${qmp1} ${qmp2} ${fifo}' RETURN EXIT
 
-	eval "$@" -chardev socket,id=mon1,path=${qmp1},server,nowait \
+	eval "$@" -chardev socket,id=mon1,path=${qmp1},server=on,wait=off \
 		-mon chardev=mon1,mode=control | tee ${migout1} &
 
 	# We have to use cat to open the named FIFO, because named FIFO's, unlike
 	# pipes, will block on open() until the other end is also opened, and that
 	# totally breaks QEMU...
 	mkfifo ${fifo}
-	eval "$@" -chardev socket,id=mon2,path=${qmp2},server,nowait \
+	eval "$@" -chardev socket,id=mon2,path=${qmp2},server=on,wait=off \
 		-mon chardev=mon2,mode=control -incoming unix:${migsock} < <(cat ${fifo}) &
 	incoming_pid=`jobs -l %+ | awk '{print$2}'`
 

From patchwork Wed Apr 28 10:18:43 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>
X-Patchwork-Id: 12228507
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-18.8 required=3.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,INCLUDES_CR_TRAILER,
	INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,USER_AGENT_GIT
	autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 1F6D7C43470
	for <kvm@archiver.kernel.org>; Wed, 28 Apr 2021 10:18:56 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id DD08661436
	for <kvm@archiver.kernel.org>; Wed, 28 Apr 2021 10:18:55 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S239255AbhD1KTj (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Wed, 28 Apr 2021 06:19:39 -0400
Received: from lindbergh.monkeyblade.net ([23.128.96.19]:34518 "EHLO
        lindbergh.monkeyblade.net" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S239233AbhD1KTi (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 28 Apr 2021 06:19:38 -0400
Received: from mail-wr1-x430.google.com (mail-wr1-x430.google.com
 [IPv6:2a00:1450:4864:20::430])
        by lindbergh.monkeyblade.net (Postfix) with ESMTPS id E610EC061574
        for <kvm@vger.kernel.org>; Wed, 28 Apr 2021 03:18:49 -0700 (PDT)
Received: by mail-wr1-x430.google.com with SMTP id m9so49686266wrx.3
        for <kvm@vger.kernel.org>; Wed, 28 Apr 2021 03:18:49 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=linaro.org; s=google;
        h=from:to:cc:subject:date:message-id:in-reply-to:references
         :mime-version:content-transfer-encoding;
        bh=h0nQnWfRsJ4H9fZ4V8oB0e3t3FwopAIx7BelND6opc4=;
        b=bXzFmQP1exjyRE4ikU7o5JEDc/TxB2cHiu9Lh59XBAJr+A3q4lARBvXmVMzRNrMj5L
         vjQJBVkXOC65HWb6w/qD1dmG8PVFdbanqJ2laZYgQq13ex4X9vXw+PPX/Zb7R3PrC+Ad
         R5Awbdh6Ev+31y58S3A+X3ushs8Rd1m6WW8Sgw4QMJc0hPQ5X+ky1FvjHRFvWutikyqN
         FmckXdM8YadSOzFWOHXv/ch+cwdB0Ghq5vu7RrFMAYpRW3zkAHvZLuRPyOVSZWtnTYrU
         J1mHfdZGgz+Sq1mGKTs8nq3/sqUOHcrxOS9ChhVQKsS+WoN9knT9GVVSSRh6Rianlb7V
         ipTA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=h0nQnWfRsJ4H9fZ4V8oB0e3t3FwopAIx7BelND6opc4=;
        b=DXf+kpwEbTETBj7Bft2Dn0+qYEJZXtNnMxeCtJa+xlJMdOPaJ/IS3e5/2VT1/Qqakd
         rMqYeoqY5q4PWNBLda2Iy7CVNWKGVWfs+KNgSyel+Nh3GnxynJ+m1leqg27xJ8pZqfVn
         HQoU6t9+SBpIKRFLyzNqdAO3mNWGzIHAQB0APwwWpauoTpPFyN9yxn0aveC7h0HQpK4k
         nt169xua3AER2/9GevOmJAW1G8GqPDf8q4jlDjbfIUpYNcbYiJDeRVQ31/rXB7msGt6N
         ZweqzWAGubVNpaPmNzcKZ7jOZJmQnHwPOu6wKK9iB/A+qJecNOyI/31Oyg62WUtyMvpE
         e93A==
X-Gm-Message-State: AOAM531e5zJxYPnEo/We4TecH2jn+afzYoRPP3Y7BK3tNthn6nXasPH7
        p58F4oBiokD3EG2I2YcDT//7uA==
X-Google-Smtp-Source: 
 ABdhPJyF87c/fp1iN0zpxYpXE/1Gh0VymotpoSflP0pBM80ElDhXfzZRhRVKwFGRvGOg2YP67jblyQ==
X-Received: by 2002:a05:6000:1564:: with SMTP id
 4mr35151201wrz.197.1619605128646;
        Wed, 28 Apr 2021 03:18:48 -0700 (PDT)
Received: from zen.linaroharston ([51.148.130.216])
        by smtp.gmail.com with ESMTPSA id
 g5sm7632844wrq.30.2021.04.28.03.18.44
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Wed, 28 Apr 2021 03:18:44 -0700 (PDT)
Received: from zen.lan (localhost [127.0.0.1])
        by zen.linaroharston (Postfix) with ESMTP id 410151FF8F;
        Wed, 28 Apr 2021 11:18:44 +0100 (BST)
From: =?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>
To: kvm@vger.kernel.org
Cc: shashi.mallela@linaro.org, alexandru.elisei@arm.com,
 eric.auger@redhat.com, qemu-arm@nongnu.org,
 linux-arm-kernel@lists.infradead.org, kvmarm@lists.cs.columbia.edu,
 christoffer.dall@arm.com, maz@kernel.org,
 =?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>
Subject: [kvm-unit-tests PATCH v1 3/4] arm64: enable its-migration tests for
 TCG
Date: Wed, 28 Apr 2021 11:18:43 +0100
Message-Id: <20210428101844.22656-4-alex.bennee@linaro.org>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20210428101844.22656-1-alex.bennee@linaro.org>
References: <20210428101844.22656-1-alex.bennee@linaro.org>
MIME-Version: 1.0
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

With the support for TCG emulated GIC we can also test these now. You
need to call run_tests.sh with -a to trigger the
its-migrate-unmapped-collection test which obviously doesn't need the
KVM errata to run in TCG system emulation mode.

Signed-off-by: Alex Bennée <alex.bennee@linaro.org>
---
 arm/unittests.cfg | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/arm/unittests.cfg b/arm/unittests.cfg
index c72dc34..d4dbc8b 100644
--- a/arm/unittests.cfg
+++ b/arm/unittests.cfg
@@ -203,7 +203,6 @@ arch = arm64
 [its-migration]
 file = gic.flat
 smp = $MAX_SMP
-accel = kvm
 extra_params = -machine gic-version=3 -append 'its-migration'
 groups = its migration
 arch = arm64
@@ -211,7 +210,6 @@ arch = arm64
 [its-pending-migration]
 file = gic.flat
 smp = $MAX_SMP
-accel = kvm
 extra_params = -machine gic-version=3 -append 'its-pending-migration'
 groups = its migration
 arch = arm64
@@ -219,7 +217,6 @@ arch = arm64
 [its-migrate-unmapped-collection]
 file = gic.flat
 smp = $MAX_SMP
-accel = kvm
 extra_params = -machine gic-version=3 -append 'its-migrate-unmapped-collection'
 groups = its migration
 arch = arm64

From patchwork Wed Apr 28 10:18:44 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>
X-Patchwork-Id: 12228505
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-18.8 required=3.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,INCLUDES_CR_TRAILER,
	INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,USER_AGENT_GIT
	autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id BC121C43462
	for <kvm@archiver.kernel.org>; Wed, 28 Apr 2021 10:18:54 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 8CE4461434
	for <kvm@archiver.kernel.org>; Wed, 28 Apr 2021 10:18:54 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S239229AbhD1KTi (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Wed, 28 Apr 2021 06:19:38 -0400
Received: from lindbergh.monkeyblade.net ([23.128.96.19]:34524 "EHLO
        lindbergh.monkeyblade.net" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S236343AbhD1KTh (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 28 Apr 2021 06:19:37 -0400
Received: from mail-wr1-x429.google.com (mail-wr1-x429.google.com
 [IPv6:2a00:1450:4864:20::429])
        by lindbergh.monkeyblade.net (Postfix) with ESMTPS id 74205C06138A
        for <kvm@vger.kernel.org>; Wed, 28 Apr 2021 03:18:52 -0700 (PDT)
Received: by mail-wr1-x429.google.com with SMTP id d11so4582065wrw.8
        for <kvm@vger.kernel.org>; Wed, 28 Apr 2021 03:18:52 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=linaro.org; s=google;
        h=from:to:cc:subject:date:message-id:in-reply-to:references
         :mime-version:content-transfer-encoding;
        bh=oZnOV6h4qE9XN8eV45ogIXJuOiGbasz5bo3bp014tJU=;
        b=ANukVHXncbLdbgHdhFWXicL6oAdfFtCcf2YirwriVA4no99pPgstEO2RBPIjibx6g8
         nlffMe5pHxW6pOhzQoEmoo/n+D4hKxlvyGOY7k+NC3PlSWu5Jb4CY8BANgzKFMZqavfU
         roWKWKJGmaCbeAun1KO/Luh2ACBlD7nxDsp0vFvUeCj+iDoPZir35+uMZzC1Cp2sr4Lu
         VVN0vSbiol6mSi5ECjnJjcXI4+hSUIkgBUI6ay7IUnvinESDf1WTMc8u1938koOC85Gy
         x99mUjPVIg6Xste2niTYJqDR5SNkeOpLFtOE6U8z8c8zJuW1KL7geU/8NzMdg+WB77n8
         gLoQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=oZnOV6h4qE9XN8eV45ogIXJuOiGbasz5bo3bp014tJU=;
        b=A0ueESOhXl/pfv8yTq620A18Mo3xMi1mX429WOBOxhxdqfAkflPGCqEp7upgsrS72v
         XWZZ9dGFg88UnPBvNc1r/AC5x2wv49NieSHQvOzZ9Gv4579fHr0Wp62DKV9Qf+kAvrpV
         VAJlqhlbkWQASY7IW7+zvKi9ZNjzAY4khAQfQqlXgN7ZikyVsAqxjwuHC04yUI6nel8x
         e8bYsAxVqhkaRYYbZ6hzeWt6pu9t2ka4pE+XwwtWKVI/CtGZiOandYqFITMIRB1D6bG6
         1R2QbsvtY3lWtPjmzzxefnl86U2wIcsfnLPlMlt+4P6A/0E4sP9ncsgg33V/MKMBr5lj
         d1qA==
X-Gm-Message-State: AOAM530BdGyKsLnsn3XjgoiG6OZe6yPImg8JFOpNhjGUddaNHCsozkRy
        Zt+hZ8hBBBGRm5+Cyut2X40sSQ==
X-Google-Smtp-Source: 
 ABdhPJx4lXogaQ0G+Fi1V4EswMRVo+Z1RplrLBVXheFEI7EdeYbid5vv/TMhgbDmRt2wHswrTJ4YPg==
X-Received: by 2002:adf:dd52:: with SMTP id u18mr2611724wrm.32.1619605131119;
        Wed, 28 Apr 2021 03:18:51 -0700 (PDT)
Received: from zen.linaroharston ([51.148.130.216])
        by smtp.gmail.com with ESMTPSA id
 u2sm5734412wmc.22.2021.04.28.03.18.44
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Wed, 28 Apr 2021 03:18:44 -0700 (PDT)
Received: from zen.lan (localhost [127.0.0.1])
        by zen.linaroharston (Postfix) with ESMTP id 4D0661FF90;
        Wed, 28 Apr 2021 11:18:44 +0100 (BST)
From: =?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>
To: kvm@vger.kernel.org
Cc: shashi.mallela@linaro.org, alexandru.elisei@arm.com,
 eric.auger@redhat.com, qemu-arm@nongnu.org,
 linux-arm-kernel@lists.infradead.org, kvmarm@lists.cs.columbia.edu,
 christoffer.dall@arm.com, maz@kernel.org,
 =?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>
Subject: [kvm-unit-tests PATCH v1 4/4] arm64: split
 its-migrate-unmapped-collection into KVM and TCG variants
Date: Wed, 28 Apr 2021 11:18:44 +0100
Message-Id: <20210428101844.22656-5-alex.bennee@linaro.org>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20210428101844.22656-1-alex.bennee@linaro.org>
References: <20210428101844.22656-1-alex.bennee@linaro.org>
MIME-Version: 1.0
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

When running the test in TCG we are basically running on bare metal so
don't rely on having a particular kernel errata applied.

You might wonder why we handle this with a totally new test name
instead of adjusting the append as we have before? Well the
run_migration shell script uses eval "$@" which unwraps the -append
leading to any second parameter being split and leaving QEMU very
confused and the test hanging. This seemed simpler than re-writing all
the test running logic in something sane ;-)

Signed-off-by: Alex Bennée <alex.bennee@linaro.org>
Cc: Shashi Mallela <shashi.mallela@linaro.org>
---
 arm/gic.c         |  7 ++++++-
 arm/unittests.cfg | 11 ++++++++++-
 2 files changed, 16 insertions(+), 2 deletions(-)

diff --git a/arm/gic.c b/arm/gic.c
index 96a329d..3bc7477 100644
--- a/arm/gic.c
+++ b/arm/gic.c
@@ -843,7 +843,7 @@ static void test_migrate_unmapped_collection(void)
 		goto do_migrate;
 	}
 
-	if (!errata(ERRATA_UNMAPPED_COLLECTIONS)) {
+	if (!errata(ERRATA_UNMAPPED_COLLECTIONS) && !under_tcg) {
 		report_skip("Skipping test, as this test hangs without the fix. "
 			    "Set %s=y to enable.", ERRATA_UNMAPPED_COLLECTIONS);
 		test_skipped = true;
@@ -1017,6 +1017,11 @@ int main(int argc, char **argv)
 		report_prefix_push(argv[1]);
 		test_migrate_unmapped_collection();
 		report_prefix_pop();
+	} else if (!strcmp(argv[1], "its-migrate-unmapped-collection-tcg")) {
+		under_tcg = true;
+		report_prefix_push(argv[1]);
+		test_migrate_unmapped_collection();
+		report_prefix_pop();
 	} else if (strcmp(argv[1], "its-introspection") == 0) {
 		report_prefix_push(argv[1]);
 		test_its_introspection();
diff --git a/arm/unittests.cfg b/arm/unittests.cfg
index d4dbc8b..e8f2e74 100644
--- a/arm/unittests.cfg
+++ b/arm/unittests.cfg
@@ -214,13 +214,22 @@ extra_params = -machine gic-version=3 -append 'its-pending-migration'
 groups = its migration
 arch = arm64
 
-[its-migrate-unmapped-collection]
+[its-migrate-unmapped-collection-kvm]
 file = gic.flat
 smp = $MAX_SMP
+accel = kvm
 extra_params = -machine gic-version=3 -append 'its-migrate-unmapped-collection'
 groups = its migration
 arch = arm64
 
+[its-migrate-unmapped-collection-tcg]
+file = gic.flat
+smp = $MAX_SMP
+accel = tcg
+extra_params = -machine gic-version=3 -append 'its-migrate-unmapped-collection-tcg'
+groups = its migration
+arch = arm64
+
 # Test PSCI emulation
 [psci]
 file = psci.flat
